### ISSUE TITLE
select (match) group by

### ISSUE THREAD

**Issue by santo-it** [orientechnologies/orientdb#7215]
## Expected behavior  
results are grouped by the specified field

## Actual behavior  
results are not grouped and and 2.2 returned recordset is cut

## Steps to reproduce  

demodb 

#initial match: works
```
match 
{
	Class: Orders, as: order, where:(Id=2)
}--{
	Class: Customers, as: customer
}-HasVisited->{
	class: Locations, as: location
} return customer, location
```

#select + match: works
```
select 
	location.Type as mytype, location.Id as myid 
from (

	match 
	{
		Class: Orders, as: order, where:(Id=2)
	}--{
		Class: Customers, as: customer
	}-HasVisited->{
		class: Locations, as: location
	} return customer, location

)
```

#select + match + group by: does not works
```
select 
	location.Type as mytype, location.Id as myid 
from (

	match 
	{
		Class: Orders, as: order, where:(Id=2)
	}--{
		Class: Customers, as: customer
	}-HasVisited->{
		class: Locations, as: location
	} return customer, location

) group by mytype
```

**Comment by luigidellaquila** [comment]
Hi Santo,

I doubt that the MATCH has anything to do with this... anyway, do you have a dataset to reproduce this?

The query in general does not make much sense, it would if you had an aggregate function for myid...

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Ok, I managed to create a basic case for this:

```
create class aaa

insert into aaa set name = 'a', surname = 'b'
insert into aaa set name = 'a', surname = 'c'

select name as p, surname as q from aaa group by surname

select name as p, surname as q from aaa group by q
```

It's not a bug, it's how the executor works in v. 2.2: the aggregate projections (thus the GROUP BY) are calculated before the aliases are assigned, so you cannot use aliases in GROUP BY (or better, you can, but you won't get what you want).

It's already fixed in develop branch (v 3.0)

Thanks

Luigi

**Comment by santo-it** [comment]
Hi Luigi,

Thanks. Sorry for late reply. The dataset is the demo db (can be easily imported into 2.2. by copying / pasting commands from https://github.com/orientechnologies/orientdb/blob/develop/distribution/src/test/resources/demoDB.osql)

About the "sense" of the query: I didn't paste the count(*) in my previous message. What I would like to do is to count the different types of locations that have been visited by a specific customer, but this query was returning an "odd" resultset imho:

```
select 
	count(*) as count, location.Type as mytype
from (

	match 
	{
		Class: Orders, as: order, where:(Id=2)
	}--{
		Class: Customers, as: customer
	}-HasVisited->{
		class: Locations, as: location
	} return customer, location

) group by mytype
```

I'll test again. Thanks
